### PR TITLE
Adds Publisher.prefix(while:behavior:).

### PR DESCRIPTION
**Pull request by jasdev**
Figured we should add this to Ext after stumbling upon @jessegrosjean’s [Swift Forums thread](https://forums.swift.org/t/how-to-make-combines-prefix-operator-inclusive/39216) and @mayoff’s [approach with composed operators](https://forums.swift.org/t/how-to-make-combines-prefix-operator-inclusive/39216/8).

### CODE REVIEW

File: Sources/Operators/PrefixWhileBehavior.swift [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        case .end:
+            return false
+        case .whileValueOrIncluded:
+            return true
+        }
+    }
+
+    var value: Output? {
+        switch self {
+        case .end:
```
Comment (freak4pc) [inline Sources/Operators/PrefixWhileBehavior.swift:71]: this abstraction confuses me a bit - can you explain?

**Comment by mayoff** [comment]
The goal is an operator that passes through all of the upstream values, until and **including** the first value that fails the predicate.

In the linked Swift forums thread, I implemented the requested operator as follows:

- Turn each upstream value into a publisher that emits the value (in a wrapper), followed by an end-marker if the value fails the predicate. In my original version, I used `Sequence.Publisher`. In @jasdev's code, the `Just ... append ... setFailureType` expression produces these short publishers.

- Use `flatMap` to flatten those short publishers into one publisher than republishes each upstream value, with an end-marker after each upstream value that fails the predicate.

- Use `prefix(while:)` to terminate the stream after the first end-marker.

- Use `compactMap` to unwrap the values and discard the end-marker.

In my original code, the wrappers and the end-marker were cases of `enum Cut`. Here, `@jasdev` has renamed `enum Cut` to `enum PrefixInclusiveEvent`.

File: Sources/Operators/PrefixWhileBehavior.swift (hunk 1 above)
Comment (mayoff) [reply Sources/Operators/PrefixWhileBehavior.swift:71]: See [my comment](https://github.com/CombineCommunity/CombineExt/pull/70#issuecomment-792062132) below.

**Comment by jasdev** [comment]
(Cleaning up some ol’ PRs — @freak4pc any chance we could land this in the next release whenever you get a chance to review? 🙏🏽)

**Review by freak4pc — APPROVED** [review]
Sorry @jasdev & @mayoff,
Barely had any time lately and this totally slipped my mind. Thanks for the ping (and explanation), looks great.

**Pull request merged**